src/plots/ternary/ternary.js

```diff
@@ -257,6 +285,9 @@ proto.adjustLayout = function(ternaryLayout, graphSize) {
     _this.clipDef.select('path').attr('d', triangleClip);
     _this.layers.plotbg.select('path').attr('d', triangleClip);
 
+    var triangleClipRelative = 'M0,' + h + 'h' + w + 'l-' + (w / 2) + ',-' + h + 'Z';
+    _this.clipDefRelative.select('path').attr('d', triangleClipRelative);
```

Comment:
Thanks for tip. This simplified a lot of `plot.js` level logic. Done in 0496144bcdff6c31e7c5a81952f8bc76d048d5cd